src/Tizen.Account.OAuth2/Tizen.Account.OAuth2/Authorizer.cs

```diff
@@ -98,83 +103,100 @@ private TokenResponse GetRefreshToken(IntPtr requestHandle)
             IntPtr error = IntPtr.Zero;
-            TokenResponse response = null;
-            Interop.Manager.Oauth2RefreshTokenCallback accessTokenCb = (IntPtr responseHandle, IntPtr usrData) =>
+            IntPtr id = IntPtr.Zero;
+            lock (_taskResponseMap)
+            {
+                id = (IntPtr)_responseCompletionId++;
+            }
+            TaskCompletionSource<TokenResponse> tcsaccessTokenResponse = new TaskCompletionSource<TokenResponse>();
+            _taskResponseMap[id] = tcsaccessTokenResponse;
+            ret = Interop.Manager.RefreshAccessToken(_managerHandle, requestHandle, _accessTokenCb, id);
+             if (ret != (int)OAuth2Error.None || error != IntPtr.Zero)
             {
-                Interop.Response.GetError(responseHandle, out error);
                 if (error != IntPtr.Zero)
                 {
-                    Log.Error(ErrorFactory.LogTag, "Error occured");
+                    throw ErrorFactory.GetException(error);
                 }
                 else
                 {
-                    IntPtr accessToken;
-                    ret = Interop.Response.GetAccessToken(responseHandle, out accessToken);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Interop failed");
-                        throw ErrorFactory.GetException(ret);
-                    }
-
-                    IntPtr tokenType;
-                    ret = Interop.Response.GetTokenType(responseHandle, out tokenType);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get token type");
-                    }
-
-                    long expiresIn;
-                    ret = Interop.Response.GetExpiresIn(responseHandle, out expiresIn);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get expires in");
-                    }
-
-                    IntPtr refreshToken;
-                    ret = Interop.Response.GetRefreshToken(responseHandle, out refreshToken);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Interop failed");
-                        throw ErrorFactory.GetException(ret);
-                    }
-
-                    IntPtr scope;
-                    ret = Interop.Response.GetScope(responseHandle, out scope);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get scope");
-                    }
+                    Log.Error(ErrorFactory.LogTag, "Interop failed");
+                    throw ErrorFactory.GetException(ret);
+                }
+            }
+            tcsaccessTokenResponse.Task.ConfigureAwait(false);
```

Comment:
you should use await

Reply:
actually GetRefreshToken() is normal function. await is used at the calling place of this function.
91:  return await Task.Run(() => GetRefreshToken(requestHandle));

Reply:
You still does not understand `Task` and `await` mechanism. 
`91: return await Task.Run(() => GetRefreshToken(requestHandle));`
It just wait end of GetRefreshToken. 
But, `private TokenResponse GetRefreshToken(IntPtr requestHandle)` is not async method
So, it does not wait a call of native callback. 

You should wait a call of native callback. but you does not

`private TokenResponse GetRefreshToken(IntPtr requestHandle)` 
It should be changed like below
`private Task<TokenResponse> GetRefreshToken(IntPtr requestHandle)`

Reply:
changes are done.